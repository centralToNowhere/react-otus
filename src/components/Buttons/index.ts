export { ButtonGameControl } from "./ButtonGameControl";
export { Button } from "./Button";
