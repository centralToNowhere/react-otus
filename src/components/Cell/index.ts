export { Cell } from "./Cell";
export type { ICell, ICellProps } from "./Cell";
