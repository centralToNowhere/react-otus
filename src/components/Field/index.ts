export { Field } from "./Field";
export type { IFiguresMap, IFieldProps } from "./Field";
