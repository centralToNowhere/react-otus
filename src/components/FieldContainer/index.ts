export {
  FieldContainer,
  createFormKey,
  getRandomCells,
  getGameCycleTimeout,
} from "./FieldContainer";
export type { FieldContainerProps } from "./FieldContainer";
