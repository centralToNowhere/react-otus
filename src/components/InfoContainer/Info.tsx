import React, { FC } from "react";
import styled from "@emotion/styled";
import { l10n } from "@/l10n/ru";

export interface IInfoProps {
  generationNumber: number;
}

export const Info: FC<IInfoProps> = ({ generationNumber }) => {
  return (
    <StyledInfoContainer>
      <p>
        {l10n.generation}: {generationNumber}
      </p>
    </StyledInfoContainer>
  );
};

const StyledInfoContainer = styled.div`
  padding: 20px;
  font-size: 1.5em;
`;
