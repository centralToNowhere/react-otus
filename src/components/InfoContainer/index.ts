export { Info } from "./Info";
export { InfoContainer } from "./InfoContainer";
export type { IInfoProps } from "./Info";
