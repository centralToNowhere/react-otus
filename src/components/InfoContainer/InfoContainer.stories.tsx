import React from "react";
import { Story, Meta } from "@storybook/react";
import { IInfoProps, Info } from "@/components/InfoContainer";

export default {
  title: "Components/Info",
  component: Info,
  args: {
    generationNumber: 254,
  },
} as Meta;

const Template: Story<IInfoProps> = (args) => <Info {...args} />;

export const InfoDefault = Template;
