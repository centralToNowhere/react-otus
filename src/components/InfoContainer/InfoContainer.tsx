import React, { FC } from "react";
import { useSelector } from "react-redux";
import { selectGenerationNumber } from "@/components/GameField";
import { Info } from "@/components/InfoContainer";

export const InfoContainer: FC = () => {
  const generationNumber = useSelector(selectGenerationNumber);

  return <Info generationNumber={generationNumber} />;
};
