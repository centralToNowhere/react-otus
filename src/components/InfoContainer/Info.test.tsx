import React from "react";
import { render, screen } from "@testing-library/react";
import { l10n } from "@/l10n/ru";
import { Info } from "@/components/InfoContainer";

describe("InfoContainer", () => {
  it("should render Info", () => {
    const { asFragment } = render(<Info generationNumber={5} />);

    const generationInfo = screen.getByText(`${l10n.generation}: 5`);

    expect(asFragment()).toMatchSnapshot();
    expect(generationInfo).toBeInTheDocument();
  });
});
