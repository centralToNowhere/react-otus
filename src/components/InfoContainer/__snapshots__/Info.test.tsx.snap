// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoContainer should render Info 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 20px;
  font-size: 1.5em;
}

<div
    class="emotion-0 emotion-1"
  >
    <p>
      Поколение: 5
    </p>
  </div>
</DocumentFragment>
`;
